project.clj

```diff
@@ -5,16 +5,16 @@
             :url  "http://www.opensource.org/licenses/mit-license.php"
             :distribution :repo}
   :min-lein-version "2.0.0"
-  :dependencies [[org.clojure/math.combinatorics "0.1.1"]
+  :dependencies [[org.clojure/math.combinatorics "0.1.3"]
                  [robert/hooke "1.3.0"]
-                 [clj-http "2.0.0"] ;; Needed here or lein might not compile clj-http first, resulting in broken builds
-                 [ring/ring-codec "1.0.0"]]
+                 [clj-http "2.2.0"] ;; Needed here or lein might not compile clj-http first, resulting in broken builds
```

Comment:
`clj-http` 3.x is out and `clj-http-fake` seems to work with it as well (tests pass).

So I think we can update this dependency to 3.1.0 and split the `:newest` profile into `:2.x` and `:3.x`. Similarly, the `test-newest` alias would also get split into `test-2.x` and `test-3.x`.